custom_components/tahoma/light.py

```diff
@@ -131,26 +134,27 @@ def effect(self) -> str:
 
     def update(self):
         """Fetch new state data for this light."""
-
         if self.should_wait():
             self.schedule_update_ha_state(True)
             return
 
         self.controller.get_states([self.tahoma_device])
 
-        if "core:LightIntensityState" in self.tahoma_device.active_states:
-            self._brightness = self.tahoma_device.active_states.get(
-                "core:LightIntensityState"
-            )
+        states = self.tahoma_device.active_states
 
-        if self.tahoma_device.active_states.get("core:OnOffState") == "on":
-            self._state = True
-        else:
-            self._state = False
+        if CORE_LIGHT_INTENSITY_STATE in states:
+            self._brightness = states.get(CORE_LIGHT_INTENSITY_STATE)
+
+        if CORE_ON_OFF_STATE in states:
+            self._state = states.get(CORE_ON_OFF_STATE) == "on"
```

Comment:
IMO, the condition are useless. Directly get the value and store them in the property. If not existing, _brightness will be None and _state False.